### PR TITLE
use sizeof for snprintf

### PR DESCRIPTION
**Pull request by neheb**
Signed-off-by: Rosen Penev <rosenp@gmail.com>

### CODE REVIEW

**Comment by whyman** [comment]
Could we not use fmt?

**Comment by neheb** [comment]
> Could we not use fmt?

I have no idea honestly. src/main uses C functions everywhere. I don't know the reason.

**Comment by KarlStraussberger** [comment]
I know: nobody changed it since the first days of mediatomb

**Comment by neheb** [comment]
Replaced with fmt. The whole codeblock looks like it could be replaced with something simpler.

Anyway, this is good enough for now I think.

File: src/main.cc [hunk 1]
```diff
@@ -342,15 +342,13 @@ int main(int argc, char** argv, char** envp)
...
-            char pidstr[20];
-            if (0 > snprintf(pidstr, 18, "%d", pid)) {
+            if (pid <= 1) {
                 log_error("Could not determine pid of running process.");
-                exit(EXIT_FAILURE);
+                std::exit(EXIT_FAILURE);
             }
 
-            // add a newline
-            strcat(pidstr, "\n");
```
Comment (KarlStraussberger) [inline src/main.cc:353]: You need to add the newline to the output to retain the file format

**Review by KarlStraussberger — CHANGES_REQUESTED** [review]

**Review by KarlStraussberger — APPROVED** [review]

**Pull request merged**